databuilder/sqlalchemy_utils.py

```diff
@@ -27,6 +33,9 @@ def is_predicate(clause):
         return True
     if isinstance(clause, BinaryExpression):
         return clause._is_implicitly_boolean
+    if isinstance(clause, UnaryExpression):
+        # Handles `NOT ...` expressions
+        return clause.operator is operators.inv
```

Comment:
note: I struggled to look up the definition of `operators.inv` and am not familiar with the SQLAlchemy source. Searching the SQLAlchemy docs for it wasn't helpeful.

I believe it's in this [compiler.py](https://github.com/zzzeek/sqlalchemy/blob/67ba6357e4e6dc8619ac63ce1e8f65310b6595bc/lib/sqlalchemy/sql/compiler.py#L271).

Reply:
Ah, slightly confusingly I think it actually comes from the stdlib `operator` module (which is why grepping for `def inv` in the SQLAlchemy source won't find anything). But I'm just treating it as an opaque marker for "the thing SQLAlchemy uses when constructing a `NOT` query", which is why I import it from `sqlalchemy.sql.elements`.

Reply:
@StevenMaude I've just tweaked this slightly. Is the new version any clearer?

Reply:
This is fine; sorry for late reply!